solutions/4.js

```diff
@@ -0,0 +1,9 @@
+const printAppleFive = (string, times, str='') =>{
+  if(times === 0){
+    return str;
+  }
+    str = str + string + ' ';
+  return printAppleFive(string, times-1, str);
```

Comment:
you don't need to update str. just pass it in! `printAppleFive(string, times-1, str+string+'');`